### ISSUE TITLE
Manual annotations are being shown in the incorrect scaffold and Additional, "interim" copies of gene models are also appearing

### ISSUE THREAD

**Issue by monicacecilia** [GMOD/Apollo#534]
### 1) Manual annotations are being shown in the incorrect scaffold. 
As I was testing the conditions described in #531 using the Honey2 genome data on Apollo2 (http://icebox.lbl.gov/Apollo2/annotator/index) I realized that after conducting a sequence alteration Apollo brought annotations from other scaffolds into the one I was working at. 
* Working in scaffold 1.33. 
* Drag gene model GB40033 into UcA area
* Create a 'Genomic Deletion' of length = 2 on coordinate 358,600
* three additional gene models were brought into the window. These gene models were created yesterday by @nathandunn. 

For example a gene model labeled with gene Name 12537555-7f50-4e78-93ac-3d8a21747f56-genea (and unique identifier 4e082d0e-8c74-4ee2-a192-c19a81c384f5) was visible (along with others) in Group1.33 a minute or two after performing the sequence alteration edit. 

![screen shot 2015-08-11 at 4 31 12 pm](https://cloud.githubusercontent.com/assets/1714621/9213857/ff05f164-404b-11e5-9230-61f2fda8a60d.png)

I retrieved the amino acid sequence, which included erroneous coordinates in the def line - in this case "Group1.33" - which is where it appeared, but not where the sequence is located.
![screen shot 2015-08-11 at 5 12 04 pm](https://cloud.githubusercontent.com/assets/1714621/9213884/5172e042-404c-11e5-9ca0-14e305fae8ac.png)
 
A BLAT search for this amino acid sequence shows that the sequence is not present in Group1.33:
![screen shot 2015-08-11 at 4 33 58 pm](https://cloud.githubusercontent.com/assets/1714621/9213905/87bcda22-404c-11e5-814c-fc9b4ebe82f9.png)

but in Group1.10:
![screen shot 2015-08-11 at 4 35 41 pm](https://cloud.githubusercontent.com/assets/1714621/9213910/95886842-404c-11e5-91f7-1e0cd734b767.png)

BUT, the UcA area in Group1.10 does not contain any gene models: 
![screen shot 2015-08-11 at 5 16 22 pm](https://cloud.githubusercontent.com/assets/1714621/9213935/c85d2c8a-404c-11e5-9f3a-b9278659a9c1.png)

and the annotations are not listed in the "Annotations" tab on the "Annotator Panel". 
![screen shot 2015-08-11 at 5 16 48 pm](https://cloud.githubusercontent.com/assets/1714621/9213937/d279f0cc-404c-11e5-98a9-6e678a116130.png)

===========================

### 2) Additional, "interim" copies of gene models are also appearing.

In a second, similar case, additional copies of the gene models I dragged into UcA appeared, with different names (as though they are partial copies of these), and with ownership of @nathandunn, with date of creation 2015-08-10 (yesterday) at 9:09 AM. The problem only seems to appear after sequence alterations, not other edits.
For example:
* In group 1.15, drag GB48991, GB40824, GB48969
* Then, drag only the first exon (on 5' end) of GB48992 and add it (merge it) to GB48991.
* Create 'Genomic Deletion' of length = 2 at coordinate 717,661
* The result are the gene models you dragged, plus some "intermediate states" for the same. No "foreign" gene models observed.

![screen shot 2015-08-11 at 5 20 03 pm](https://cloud.githubusercontent.com/assets/1714621/9214007/942ff252-404d-11e5-8228-003607cce4b6.png)

**Comment by nathandunn** [comment]
Thanks you for being thorough . . . another early morning for me

**Comment by nathandunn** [comment]
I'm wondering if this is just garbage data from adding via a ```add_transcripts_form_gff3_to_annotations.pl```, which would obviously also need to be fixed.

**Comment by nathandunn** [comment]
I think that the problem is that the websocket is ghosting the bad data on some operations.  Solution, dump services that create bad data.

**Comment by nathandunn** [comment]
The problem was a bad overlapper function.  Fixing it also massively sped up the operation, as well.

**Comment by monicacecilia** [comment]
:confused: ... nope. still happening.... I can see annotations you made on August 9, 10th, and 12th. 
![screen shot 2015-08-12 at 3 22 52 pm](https://cloud.githubusercontent.com/assets/1714621/9238278/02b18714-4106-11e5-829a-6ab27412b050.png)

**Comment by nathandunn** [comment]
it is RC6, I have not deployed this yet.

**Comment by monicacecilia** [comment]
neever miiiind... I just realized icebox only has RC-5...

**Comment by monicacecilia** [comment]
ah, exactly. what you said. . .

**Comment by nathandunn** [comment]
Note that I re-added the same genes (but with proper names), but it is the same script and data.

**Comment by monicacecilia** [comment]
This isn't ready :pensive:  Here is the note I added in #456 
- @nathandunn Apollo is still changing the name of the transcript without user input. In this case, after "Undo" of a boundary modification, Apollo removed the name I had given the transcript and renamed it as the first transcript (xxxgenenamexxx-00001) from that gene model. I was working on gene model GB40030 in Group1.33. This has to do with #534 - and it is still not fixed.

**Comment by nathandunn** [comment]
@monicacecilia   
1) I think that this bug can be closed if this is the only thing outstanding.
2) The "Undo" behavior is working as I designed it.  We don't (yet) track "setName", "setSymbol", etc. which is unchanged from WA1.   In WA1 they were keeping the name, however, but history functions were inconsistent, which is now fixed.  My suggestion would be to create a new issue for this and describe what things we want to retain in history.  My suggestion would be to track history for setName, setSymbol as we do for genomic features (so it would be obvious as to why your name is rechanging).   Tracking meta-info (name, symbol, description) and genomic info separately would have to be well-defined or it will be a mess.
